Assets/Scripts/Ui/Achivment.cs

```diff
@@ -0,0 +1,76 @@
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using System;
+
+namespace TankBattle.UI
+{
+    public class Achivment : MonoBehaviour
+    {
+        private Animator popUp;
+        private TextMeshProUGUI[] achievementText;
+        private Rank rank;
+        private int enemyCount = 0;
+        private int shootCount = 0;
+
+        private void Start()
+        {
+            popUp = GetComponent<Animator>();
+            achievementText = GetComponentsInChildren<TextMeshProUGUI>();
+            MVC.Enemy.TankController.OnEnemyDeath += EnemyKillAchievement;
+            MVC.Player.TankController.OnPlayerShoot += ShootingAchievement;
+        }
+
```

Comment:
For now this achievement system will work but what will happen if there are 100 types of achievements. Would you create 100 different methods for all the achievements? No, right. 

So Create a generic method which will handle all types of achievements based on the parameters you pass into it.
Take the help of your clanmates or ask the questions in discord.